### ISSUE TITLE
GatedSAEDecoder: Python and Rust use different gating predicates (numerical mismatch)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#227]
## Summary

`GatedSAEDecoder.decode` in Python and `gated_sae_decode` in Rust apply different gating predicates, so they produce different outputs for the same inputs/weights.

- **Python** (\`gamfit/_penalties.py:621\`): \`gates = (sigmoid(x @ W_gate.T) > 0.5)\` — equivalent to \`logit > 0\`.
- **Rust** (\`src/terms/gated_decoder.rs:61\`): \`if logit != 0.0 { gated[i] = x[i] }\` — gates on any non-zero logit, including negative.

The Rust comment claims this is intentional (\"gate triggers on any non-zero pre-activation, |W_gate x|_i > 0\"), but \`!= 0.0\` is almost always true for floating-point logits, making the gate a no-op in practice. The standard Gated-SAE formulation (Rajamanoharan et al. 2024) uses Heaviside(pi_gate), i.e. \`logit > 0\`, matching the Python implementation.

## Repro

\`\`\`python
import numpy as np
from gamfit import GatedSAEDecoder
from gamfit._binding import rust_module

rng = np.random.default_rng(202)
w_gate = rng.standard_normal((4, 4))
w_amp  = rng.standard_normal((6, 4))
x      = rng.standard_normal((10, 4))

py   = GatedSAEDecoder(w_gate=w_gate, w_amp=w_amp).decode(x)
rust = np.asarray(rust_module().gated_sae_decode(x, w_gate, w_amp))
np.testing.assert_allclose(py, rust)  # AssertionError
\`\`\`

## Root cause

| Side   | File:line                                | Predicate                            |
|--------|------------------------------------------|--------------------------------------|
| Python | \`gamfit/_penalties.py:621\`             | \`sigmoid(logit) > 0.5\` ≡ \`logit > 0\` |
| Rust   | \`src/terms/gated_decoder.rs:47-63\`     | \`logit != 0.0\`                     |

## Proposed fix

Align Rust to the canonical Gated-SAE Heaviside contract (\`logit > 0.0\`), which matches Python. Concretely in \`src/terms/gated_decoder.rs:61\`:

\`\`\`rust
if logit > 0.0 {
    gated[gate_row] = x[gate_row];
}
\`\`\`

Then update the comment at lines 47-54 to reflect the standard formulation. Add a parity test asserting \`GatedSAEDecoder(...).decode(x)\` matches \`rust_module().gated_sae_decode(x, w_gate, w_amp)\` to prevent regressions.

If the Rust author intended a non-standard contract, the alternative is to change Python to match Rust — but \`!= 0.0\` effectively disables gating and is unlikely to be the intent.

**Comment by SauersML** [comment]
Fixed in db4589851 — `src/terms/gated_decoder.rs:59` now gates on `logit > 0.0`, aligning with the canonical Rajamanoharan et al. 2024 Gated-SAE Heaviside contract and matching the Python (`gamfit/_penalties.py:621` — `sigmoid(logit) > 0.5`) and torch (`gamfit/torch/modules.py:330` — `gate_logits > 0`) implementations.

RED tests landed in ed6aa489e:
- `tests/gated_sae_decoder_heaviside_contract.rs` — Rust unit tests (negative / zero / positive / mixed-sign logits)
- `tests/torch/test_gated_sae_decoder_heaviside_contract.py` — Python parity tests including negative logits, zero logits, random fp64 inputs

**Verification status:** `pytest tests/torch/test_gated_sae_decoder_heaviside_contract.py::test_python_decode_matches_heaviside_reference` passes against the current Python implementation. Cross-impl parity (`cargo test --test gated_sae_decoder_heaviside_contract` and the Rust-calling Python tests) requires rebuilding the `gamfit._rust.abi3.so` extension, which is currently blocked by unrelated build.rs banned-pattern lint failures in `src/gpu/*` from concurrent work (unused `pub(crate)` items). The contract change itself is a single predicate flip and verified by source reading.
